pkg/service/wire.go

```diff
@@ -109,25 +109,54 @@ func createRedisClient(conf *config.Config) (*redis.Client, error) {
 		return nil, nil
 	}
 
-	logger.Infow("using multi-node routing via redis", "addr", conf.Redis.Address)
-	rcOptions := &redis.Options{
-		Addr:     conf.Redis.Address,
-		Username: conf.Redis.Username,
-		Password: conf.Redis.Password,
-		DB:       conf.Redis.DB,
-	}
-	if conf.Redis.UseTLS {
-		rcOptions = &redis.Options{
+	var rc *redis.Client
+	if conf.UseSentinel() {
+		logger.Infow("using multi-node routing via redis", "sentinel", true, "addr", conf.Redis.SentinelAddresses, "masterName", conf.Redis.MasterName)
+		rcOptions := &redis.FailoverOptions{
```

Comment:
can we reduce the amount of replication here? it seems that the only difference is TLS?

Reply:
Updated in the newer push.